File: bootstrap-fabric/src/main/java/ru/hollowhorizon/hollowengine/fabric/internal/DelegatedReloadListener.java
package ru.hollowhorizon.hollowengine.fabric.internal;

import net.fabricmc.fabric.api.resource.IdentifiableResourceReloadListener;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.server.packs.resources.PreparableReloadListener;
import net.minecraft.server.packs.resources.ResourceManager;
import net.minecraft.util.profiling.ProfilerFiller;

import java.util.Locale;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Executor;

public record DelegatedReloadListener(
        PreparableReloadListener eventListener) implements PreparableReloadListener, IdentifiableResourceReloadListener {
    @Override
    public String getName() {
        return eventListener.getName();
    }

    @Override
    public ResourceLocation getFabricId() {
        return ResourceLocation.fromNamespaceAndPath("hollowengine", eventListener.getClass().getName().toLowerCase(Locale.ROOT).replace('$', '.'));
    }

    @Override
    public CompletableFuture<Void> reload(PreparationBarrier preparationBarrier, ResourceManager resourceManager, ProfilerFiller profilerFiller, ProfilerFiller profilerFiller2, Executor executor, Executor executor2) {
        return eventListener.reload(preparationBarrier, resourceManager, profilerFiller, profilerFiller2, executor, executor2);
    }
}

File: src/main/java/ru/hollowhorizon/hollowengine/fabric/internal/DelegatedReloadListener.kt
//? if fabric {
/*package ru.hollowhorizon.hollowengine.fabric.internal

import net.fabricmc.fabric.api.resource.IdentifiableResourceReloadListener
import net.minecraft.resources.ResourceLocation
import net.minecraft.server.packs.resources.PreparableReloadListener
import ru.hollowhorizon.hollowengine.common.utils.rl

class DelegatedReloadListener(private val eventListener: PreparableReloadListener) :
    PreparableReloadListener by eventListener, IdentifiableResourceReloadListener {
    override fun getFabricId(): ResourceLocation {
        return "hollowengine_generated:${eventListener.javaClass.name.lowercase().replace("$", ".")}".rl
    }

    override fun getName(): String? {
        return eventListener.name
    }
}
*///?}